### ISSUE TITLE
about the number of gpus or IMS_PER_BATCH

### ISSUE THREAD

**Issue by chenz97** [zhang-tao-whu/DVIS_Plus#4]
Hello, thank you for your great work and sharing the code! 

I noticed there is a $NUM_GPUS over-writing the config values of IMS_PER_BATCH in command line options. I am wondering what is the number of gpus or IMS_PER_BATCH you used to obtain the reported performance? Do you use the default values in the config file?

I hope you can answer the question. Thank you very much!

**Comment by zhang-tao-whu** [comment]
Yes, the IMS_PER_BATCH in the config of DVIS corresponds to the number of GPUs we actually use. When training segmenter, such as MinVIS or CTVIS configs, the number of GPUs does not need to match IMS_PER_BATCH, it can be adjusted according to the available memory. The batch size has little impact on accuracy; for example, the accuracy difference between batch size 4 and batch size 8 is not significant. In fact, halve the batch size (8->4) and doubling the number of iterations may lead to better results. However, I haven't tried batch sizes of 2 or 1, so I'm not sure if it will affect convergence.

**Comment by chenz97** [comment]
Thank you for the reply! I have a further question. For training on OVIS with ViT-L for the online setting, I could not reproduce the performance you reported. When training with 4 / 8 / 16 batch size, the AP is 43.5 / 43.6 / 44.4. The paper reports an AP of 49.6.

I initialize the model with ctvis_vitl_ovis.pth and train with the following command,

```
python train_net_video.py \
  --num-gpus 8 \
  --config-file configs/dvis_Plus/ovis/vit_adapter/DVIS_Plus_Online_VitAdapterL.yaml \
  --resume MODEL.WEIGHTS ctvis_vitl_ovis.pth \
  SOLVER.IMS_PER_BATCH 8
```

I'm wondering is the command correct? Or is there any other thing I should do to reproduce the performance? Thank you very much!

**Comment by zhang-tao-whu** [comment]
This is a bit strange, because I have repeated the experiment with this version of the code, and the performance was normal. Could you provide more detailed information, such as the training log.txt?

**Comment by chenz97** [comment]
[log.txt](https://github.com/zhang-tao-whu/DVIS_Plus/files/14622942/log.txt)

Sure. I have attached the log file, which is run with a batch size of 4, using 2 RTX 3090. Thank you.

**Comment by zhang-tao-whu** [comment]
Currently, only one batch per GPU is supported, so with 2 GPUs, the batch size should be set to 2. If n video clips are placed on one GPU, they will be treated as a single video clip of length n*T. If there is overlapping ground truth IDs, it may result in erroneous ground truth.

**Comment by zhang-tao-whu** [comment]
`python train_net_video.py \
  --num-gpus 8 \
  --config-file configs/dvis_Plus/ovis/vit_adapter/DVIS_Plus_Online_VitAdapterL.yaml \
  --resume MODEL.WEIGHTS ctvis_vitl_ovis.pth \
  SOLVER.IMS_PER_BATCH 8`This command is correct. Have you tried it? What about the results? If you still encounter issues, it would be helpful if you could provide the log file corresponding to this command.

**Comment by chenz97** [comment]
I have tried num_gpus 4 and IMS_PER_BATCH 4, the performance is 47.6. You have mentioned the accuracy difference between batch size 4 and batch size 8 is not significant, I'm wondering the specific number you have obtained. I have also attached the log file below. Thank you.
[log.txt](https://github.com/zhang-tao-whu/DVIS_Plus/files/14696978/log.txt)

**Comment by zhang-tao-whu** [comment]
Hello, I have only tried using ResNet50 with batch size 4 and batch size 8, and the difference is not significant, only about 0.5 AP lower. I suggest that when the batch size is 4, you can halve the learning rate and double the number of iterations.

**Comment by chenz97** [comment]
Thank you for the response.
